### PR TITLE
Fix #4872: Implement java.util.StringJoiner.

### LINKED ISSUE #4872: Support java.util.StringJoiner
It would be nice to have support for java.util.StringJoiner, I sometimes find it useful vs mkString in imperative code.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Review by gzm0 — COMMENTED** [review]
Just some comments RE test cases.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/StringJoinerTest.scala [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      new StringJoiner(", ", "{", "}").merge(empty).setEmptyValue("+++")
+    }
+    assertJoinerResult("[one, single, two]") {
+      new StringJoiner(", ", "[", "]").add("one").merge(single).add("two")
+    }
+    assertJoinerResult("[one, a;b;c, two]") {
+      new StringJoiner(", ", "[", "]").add("one").merge(multiple).add("two")
+    }
+    assertJoinerResult("[one, single, two]") {
+      new StringJoiner(", ", "[", "]").add("one").merge(single).add("two")
```
[the remark this answers is not in the document]
Comment (sjrd) [reply test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/StringJoinerTest.scala]: Oops, no, it was meant to be a test that returns `"[single]"`. Fixed.

**Comment by sjrd** [comment]
Updated. I also added a bunch of tests with *blank* elements (empty string elements that make the joiner non-empty).

**Review by gzm0 — APPROVED** [review]
Just some minor comments RE tests.
